app/src/main/java/com/example/mytestproject/ui/searchCity/SearchCityViewModel.kt

```diff
@@ -5,35 +5,96 @@ import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.example.data.db.dao.CityDao
+import com.example.data.mappers.LastTenChosenCitiesEntityMapper
 import com.example.domain.models.CityModel
+import com.example.domain.useCase.weatherData.GetLastTenChosenCitiesUseCase
 import com.example.mytestproject.util.CITY_ID
 import com.example.mytestproject.util.CityFilter
 import com.example.mytestproject.util.Event
+import com.example.mytestproject.util.MAX_TABLE_SIZE
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
 
 class SearchCityViewModel(
     private val cityFilter: CityFilter,
-    private val sharePref: SharedPreferences
+    private val sharePref: SharedPreferences,
+    private val cityDao: CityDao,
```

Comment:
There shouldn't be any Dao in ViewModel. 
The classic chain of object is: ViewModel > UseCase > Repository > DAO/API

Reply:
Fixed